tests/test_evolution_traversal.py

```
@@ -0,0 +1,115 @@
+        # If it returns, check status code.
+        # But TestClient might raise exception directly if unhandled.
+        assert response.status_code != 200
+    except ValueError as e:
+        assert "invalid characters" in str(e)
+
+    # Ensure generate_evolution_patch was NOT called (because validation happened before)
+    assert mock_generate.call_count == 0
+
+def test_validate_patch_traversal_blocked(mock_env):
+    """Test path traversal in validate endpoint is BLOCKED."""
+    runs_root, mock_evo = mock_env
```

Ruff (0.14.14)

[warning] Unpacked variable `runs_root` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)